File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int a=0, l=0;
        int r=height.length-1;
        while(l<r){
            int w;
            if(height[l]>height[r]){
                w= (r-l)*height[r];
                r-=1;
            }
            else{
                w= (r-l)*height[l];
                l+=1;
            }
            if(w>a)
                a=w;
        }
        return a;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        a,i,j=0,0,len(height)-1
        while(i<j):
            w= (j-i) * min(height[i],height[j])
            a=max(w,a)
            if(height[i]>height[j]):
                j-=1
            else:
                i+=1
        return a